tfprotov6/tf6server/server.go

```diff
@@ -475,97 +414,101 @@ func New(name string, serve tfprotov6.ProviderServer, opts ...ServeOpt) tfplugin
 	}
 	envVar := conf.envVar
 	if envVar == "" {
-		addr, err := tfaddr.ParseRawProviderSourceString(name)
-		if err != nil {
-			log.Printf("[ERROR] Error parsing provider name %q: %s", name, err)
-		} else {
-			envVar = addr.Type
-		}
+		envVar = logging.ProviderLoggerName(name)
 	}
-	envVar = strings.ReplaceAll(envVar, "-", "_")
 	if envVar != "" {
-		options = append(options, tfsdklog.WithLogName(envVar), tflog.WithLevelFromEnv("TF_LOG_PROVIDER", envVar))
+		options = append(options, tfsdklog.WithLogName(envVar), tflog.WithLevelFromEnv(logging.EnvTfLogProvider, envVar))
 	}
 	return &server{
-		downstream:   serve,
-		stopCh:       make(chan struct{}),
-		tflogOpts:    options,
-		tflogSDKOpts: sdkOptions,
-		name:         name,
-		useTFLogSink: conf.useLoggingSink != nil,
-		testHandle:   conf.useLoggingSink,
+		downstream:      serve,
+		stopCh:          make(chan struct{}),
+		tflogOpts:       options,
+		tflogSDKOpts:    sdkOptions,
+		name:            name,
+		useTFLogSink:    conf.useLoggingSink != nil,
+		testHandle:      conf.useLoggingSink,
+		protocolDataDir: os.Getenv(logging.EnvTfLogSdkProtoDataDir),
+		protocolVersion: "6",
```

Comment:
Are there constants defined somewhere for protocol versions?

Reply:
Not at the moment, but there certainly could be in this package. Unfortunately, the protocol buffers doesn't export that particular information. Maybe that is something that could be done there, so it is always "correct" when imported.

There are two current usages of the protocol version in these server packages:

* The string here, which is the value used for logging the protocol version.
* An int in the go-plugin instantiation, which has implications with the plugin handshake.

The challenge with a single constant is that over time the string version could and probably should include the minor protocol version, should we opt into bump that with backwards compatible new features in the protocol.

Given that, I'm tempted to introduce the following constants defined at the top of these server packages:

```go
const (
  protocolVersionMajor uint = 6
  protocolVersionMinor uint = 0
  protocolVersion = fmt.Sprintf("%d.%d", protocolVersionMajor, protocolVersionMinor)
)
```

Where `protocolVersionMajor` is used in the go-plugin handshake configuration and `protocolVersion` is saved in the server here.